Yoast/javascript
# Use word stems to improve relevant words

## Summary

This PR can be summarized in the following changelog entry:

* Improves internal linking suggestions for English through normalizing relevant words with a stemming mechanism
* Removes 2,3,4,5-word prominent word combinations

## Relevant technical choices:

* It was decided to remove multiple-word word combinations from relevant words analysis because those were not efficient for internal linking suggestions while they take up a lot of computational resources.

## Test instructions

This PR can be tested by following these steps:

* Use the `example`
* Make sure specify `en_EN` locale
* Check that no 2,3,4,5-word word combinations are displayed among `Relevant words for linking suggestions` or `Relevant words for insights`.
* For `Relevant words for linking suggestions`: Check that words used in paper attributes (title, keyphrase, synonyms, metadescription, subheadings) as well as in the main text of the copy get collapsed correctly. I.e., there should not be any duplicate words in the output table and the number of occurrences for every word should be occurrences_in_text + 3*occurrences_in_attributes
* For both `Relevant words for linking suggestions` and `Relevant words for insights`: Check if a stem is displayed next to every relevant word
* For both `Relevant words for linking suggestions` and `Relevant words for insights`: Check that there are no two entries with the same stem in the table, i.e., all same-stem words were correctly collapsed. In order to test it, you can add different forms of the word in the text and in the paper attributes
*  For both `Relevant words for linking suggestions` and `Relevant words for insights`: Check that words with stems that were only used in the text once do not end up in the list
*  For both `Relevant words for linking suggestions` and `Relevant words for insights`: Check that stems that were used in the text in different forms, while at least some of the forms were only used once, do end up in the list and are counted correctly
*  For both `Relevant words for linking suggestions` and `Relevant words for insights`: Check that function words do not end up in the list

Fixes https://github.com/Yoast/YoastSEO.js/issues/2139
Transfered from YoastSEO.js (https://github.com/Yoast/YoastSEO.js/pull/2174)
